### PR TITLE
Add tests for expval, var and probs with seed

### PR DESCRIPTION
**Pull request by rauletorresc**
**Context:** Once https://github.com/PennyLaneAI/pennylane-lightning/pull/1003 gets merged into Lightning, we can test `expval`, `var` and `probs` with a seed from the Catalyst frontend.

**Description of the Change:** Add the corresponding tests.

[sc-77124]

### CODE REVIEW

File: doc/requirements.txt [hunk 1]
```diff
@@ -30,6 +30,6 @@ lxml_html_clean
 
 # Pre-install PL development wheels
 --extra-index-url https://test.pypi.org/simple/
-pennylane-lightning-kokkos==0.40.0-dev14
-pennylane-lightning==0.40.0-dev14
+pennylane-lightning-kokkos==0.40.0-dev22
+pennylane-lightning==0.40.0-dev22
```
Comment (paul0403) [inline doc/requirements.txt:34]: update to the correct testpypi version once https://github.com/PennyLaneAI/pennylane-lightning/pull/1003 is merged in lightning

Comment (rauletorresc) [reply doc/requirements.txt]: Done

File: frontend/test/pytest/test_seeded_qjit.py [hunk 2]
```diff
@@ -150,5 +150,154 @@ def circuit():
...
+
+    dev = qml.device(backend, wires=2, shots=shots)
+
+    @qjit(seed=seed)
+    def workflow():
+        @qml.qnode(dev)
+        def circuit():
+            qml.RY(0.7, wires=0)
+            qml.PauliZ(0)
+            return qml.probs()
```
Comment (paul0403) [inline frontend/test/pytest/test_seeded_qjit.py:180]: Is `state` not seeded?

File: frontend/test/pytest/test_seeded_qjit.py [hunk 3]
```diff
@@ -150,5 +150,154 @@ def circuit():
         assert np.allclose(results0, results2)
 
 
+@pytest.mark.parametrize(
+    "seed",
+    [
+        42,
+        37,
+        1337,
```
[the remark this answers is not in the document]
Comment (rauletorresc) [reply frontend/test/pytest/test_seeded_qjit.py]: Makes sense. Done.

**Review by paul0403 — APPROVED** [review]
:muscle:

**Pull request merged**